### PR TITLE
r/glue_registry - new resource

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #16342
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_glue_registry - add new resource
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGlueRegistry_'
--- PASS: TestAccAWSGlueRegistry_disappears (29.99s)
--- PASS: TestAccAWSGlueRegistry_basic (41.39s)
--- PASS: TestAccAWSGlueRegistry_Description (72.98s)
--- PASS: TestAccAWSGlueRegistry_tags (103.72s)
```

### CODE REVIEW

File: website/docs/r/glue_registry.html.markdown [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `arn` - Amazon Resource Name (ARN) of Glue Registry.
+* `id` - Amazon Resource Name (ARN) of Glue Registry.
+
+## Import
+
+Glue Registrys can be imported using `arn`, e.g.
```
Comment (ewbankkit) [inline website/docs/r/glue_registry.html.markdown:38]:
```suggestion
Glue Registries can be imported using `arn`, e.g.
```

File: aws/internal/service/glue/id.go [hunk 2]
```diff
@@ -30,3 +32,9 @@ func stringifyAwsGluePartition(partValues *schema.Set) string {
 
 	return vals
 }
+
+func CreateAwsGlueRegistryID(id string) *glue.RegistryId {
```
Comment (ewbankkit) [inline aws/internal/service/glue/id.go:36]: Nit. My understanding is that `Create` and `Parse` ID functions are meant to work on strings only, so this functionality can just be moved into `finder.RegistryByID`.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueRegistry_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueRegistry_ -timeout 120m
=== RUN   TestAccAWSGlueRegistry_basic
=== PAUSE TestAccAWSGlueRegistry_basic
=== RUN   TestAccAWSGlueRegistry_Description
=== PAUSE TestAccAWSGlueRegistry_Description
=== RUN   TestAccAWSGlueRegistry_tags
=== PAUSE TestAccAWSGlueRegistry_tags
=== RUN   TestAccAWSGlueRegistry_disappears
=== PAUSE TestAccAWSGlueRegistry_disappears
=== CONT  TestAccAWSGlueRegistry_basic
=== CONT  TestAccAWSGlueRegistry_disappears
=== CONT  TestAccAWSGlueRegistry_tags
=== CONT  TestAccAWSGlueRegistry_Description
--- PASS: TestAccAWSGlueRegistry_disappears (13.39s)
--- PASS: TestAccAWSGlueRegistry_basic (19.79s)
--- PASS: TestAccAWSGlueRegistry_Description (32.31s)
--- PASS: TestAccAWSGlueRegistry_tags (43.51s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	43.579s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2).

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueRegistry_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueRegistry_ -timeout 120m
=== RUN   TestAccAWSGlueRegistry_basic
=== PAUSE TestAccAWSGlueRegistry_basic
=== RUN   TestAccAWSGlueRegistry_Description
=== PAUSE TestAccAWSGlueRegistry_Description
=== RUN   TestAccAWSGlueRegistry_tags
=== PAUSE TestAccAWSGlueRegistry_tags
=== RUN   TestAccAWSGlueRegistry_disappears
=== PAUSE TestAccAWSGlueRegistry_disappears
=== CONT  TestAccAWSGlueRegistry_basic
=== CONT  TestAccAWSGlueRegistry_disappears
=== CONT  TestAccAWSGlueRegistry_tags
=== CONT  TestAccAWSGlueRegistry_Description
--- PASS: TestAccAWSGlueRegistry_disappears (11.29s)
--- PASS: TestAccAWSGlueRegistry_basic (15.10s)
--- PASS: TestAccAWSGlueRegistry_Description (25.43s)
--- PASS: TestAccAWSGlueRegistry_tags (35.70s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	37.355s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)  (Registries appear not to be supported in GovCloud)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueRegistry_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueRegistry_ -timeout 120m
=== RUN   TestAccAWSGlueRegistry_basic
=== PAUSE TestAccAWSGlueRegistry_basic
=== RUN   TestAccAWSGlueRegistry_Description
=== PAUSE TestAccAWSGlueRegistry_Description
=== RUN   TestAccAWSGlueRegistry_tags
=== PAUSE TestAccAWSGlueRegistry_tags
=== RUN   TestAccAWSGlueRegistry_disappears
=== PAUSE TestAccAWSGlueRegistry_disappears
=== CONT  TestAccAWSGlueRegistry_basic
=== CONT  TestAccAWSGlueRegistry_disappears
=== CONT  TestAccAWSGlueRegistry_Description
=== CONT  TestAccAWSGlueRegistry_tags
=== CONT  TestAccAWSGlueRegistry_disappears
    resource_aws_glue_registry_test.go:191: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueRegistry_disappears (0.99s)
=== CONT  TestAccAWSGlueRegistry_tags
    resource_aws_glue_registry_test.go:191: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueRegistry_tags (0.99s)
=== CONT  TestAccAWSGlueRegistry_basic
    resource_aws_glue_registry_test.go:191: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueRegistry_basic (1.00s)
=== CONT  TestAccAWSGlueRegistry_Description
    resource_aws_glue_registry_test.go:191: skipping acceptance testing: AccessDeniedException: Operation not allowed
--- SKIP: TestAccAWSGlueRegistry_Description (1.01s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2.595s
```

**Pull request merged**